mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/event/impl/OutboxPollingOutboxEventAdditionalJaxbMappingProducer.java

```diff
@@ -123,32 +89,21 @@ public final class OutboxPollingOutboxEventAdditionalJaxbMappingProducer
 					.asString()
 					.build();
 
-	@SuppressWarnings("deprecation")
-	private static final OptionalConfigurationProperty<PayloadType> ENTITY_MAPPING_OUTBOXEVENT_PAYLOAD_TYPE =
-			ConfigurationProperty.forKey(
-					HibernateOrmMapperOutboxPollingSettings.CoordinationRadicals.ENTITY_MAPPING_OUTBOXEVENT_PAYLOAD_TYPE )
-					.as( PayloadType.class, PayloadType::of )
-					.build();
 
 	@Override
-	@SuppressForbiddenApis(reason = "Strangely, this SPI involves the internal MappingBinder class,"
-			+ " and there's nothing we can do about it")
-	public Collection<MappingDocument> produceMappings(ConfigurationPropertySource propertySource, Dialect dialect,
-			MappingBinder mappingBinder, MetadataBuildingContext buildingContext) {
-
+	public Map<Class<?>, JaxbEntityMappings> produceMappings(ConfigurationPropertySource propertySource, Dialect dialect,
+			MetadataBuildingContext buildingContext) {
 		Optional<String> mapping = OUTBOXEVENT_ENTITY_MAPPING.get( propertySource );
 		Optional<String> schema = ENTITY_MAPPING_OUTBOXEVENT_SCHEMA.get( propertySource );
 		Optional<String> catalog = ENTITY_MAPPING_OUTBOXEVENT_CATALOG.get( propertySource );
 		Optional<String> table = ENTITY_MAPPING_OUTBOXEVENT_TABLE.get( propertySource );
+
 		Optional<UuidGenerationStrategy> uuidStrategy = ENTITY_MAPPING_OUTBOXEVENT_UUID_GEN_STRATEGY.get( propertySource );
 		Optional<String> uuidType = ENTITY_MAPPING_OUTBOXEVENT_UUID_TYPE.get( propertySource );
```

Comment:
Could you rather use `getAndTransform` here and call `UuidDataTypeUtils.uuidType` in the lambda? Then your error message can be much simpler as Hibernate Search will mention the property key and value automatically (among other things).

Reply:
Same for the other mapping producer, obviously.

Reply:
Hmm, I'm not sure that this will work... Since we are only logging a warning rather than throwing an exception, the logged message won't get the context of the property-being-resolved... I'll keep it the way it is for now, and we can discuss this tomorrow.

Reply:
So first, I was wrong in suggesting `getAndTransform`, it's actually `getAndMap` that you should use.

Regarding your comment, `unableToParseJdbcTypeCode` would work just fine as it's an exception, but `usingDeprecatedPropertyValue` is indeed a warning (I didn't notice that)  and would indeed still need access to the property name.

IMO you can still go with `getAndMap`, and resolve the property name if a warning really is needed. Maybe change the signature of your method to something like:

```
public static int uuidType(ConfigurationPropertySource source, OptionalConfigurationProperty property, Dialect dialect) {
```

... and work from there?

Reply:
I meant it when I suggested this signature:

```
public static int uuidType(ConfigurationPropertySource source, OptionalConfigurationProperty property, Dialect dialect) {
```

If you also pass the value, there's not much sense in doing this. The idea was that `uuidType()` itself would call `property.get(source).getAndTransform(value -> { ... })`.

Reply:
Ah, I see you called `getAndMap` in the callers... whatever, this works too. Ignore me :)